pallets/phala/src/registry.rs

```diff
@@ -337,6 +372,57 @@ pub mod pallet {
 			Ok(())
 		}
 
+		#[pallet::weight(0)]
+		pub fn upload_code(origin: OriginFor<T>, code: Vec<u8>) -> DispatchResult {
+			ensure_signed(origin)?;
+			let code_hash = T::Hashing::hash(&code.encode());
+			PristineCode::<T>::insert(&code_hash, &code);
+			Self::deposit_event(Event::CodeUploaded(code_hash));
+			Ok(())
+		}
+
+		#[pallet::weight(0)]
+		pub fn instantiate_contract(
+			origin: OriginFor<T>,
+			// #[pallet::compact] endowment: BalanceOf<T>,
+			// #[pallet::compact] gas_limit: Weight,
+			code_index: CodeIndex<CodeHash<T>>,
+			data: Vec<u8>,
+			salt: Vec<u8>,
+			deploy_worker: Option<WorkerPublicKey>,
+		) -> DispatchResult {
```

Comment:
Do we need a on-chain switch to enable the pink deployment?

Reply:
for now, this intends to be only available in testnet, shall add one for mainnet for compatiablility

Reply:
Sounds good to me